src/components/game.js

```diff
@@ -36,21 +36,152 @@ class Game extends React.Component {
+                ground[i][k - 1].value = 0;
+                break;
+              } else if (ground[i][k].value === 0) {
+                ground[i][k].value = ground[i][k - 1].value;
+                ground[i][k - 1].value = 0;
+              }
+            }
+          }
+        }
+      }
+    } else if (direction.key === 'ArrowLeft') {
+      for (i = 0; i < data.site.siteMetadata.size; i += 1) {
+        for (j = 1; j < data.site.siteMetadata.size; j += 1) {
+          if (ground[i][j].value !== 0) {
+            for (k = j - 1; k >= 0; k -= 1) {
+              if (ground[i][k + 1].value === ground[i][k].value && ground[i][k].isUsed === false) {
+                ground[i][k].value *= 2;
+                ground[i][k].isUsed = true;
+                ground[i][k + 1].value = 0;
+                break;
+              } else if (ground[i][k].value === 0) {
+                ground[i][k].value = ground[i][k + 1].value;
+                ground[i][k + 1].value = 0;
+              }
+            }
+          }
+        }
+      }
+    } else if (direction.key === 'ArrowUp') {
+      for (j = 0; j < data.site.siteMetadata.size; j += 1) {
+        for (i = 1; i < data.site.siteMetadata.size; i += 1) {
+          if (ground[i][j].value !== 0) {
+            for (k = i - 1; k >= 0; k -= 1) {
+              if (ground[k + 1][j].value === ground[k][j].value && ground[k][j].isUsed === false) {
+                ground[k][j].value *= 2;
+                ground[k][j].isUsed = true;
+                ground[k + 1][j].value = 0;
+                break;
+              } else if (ground[k][j].value === 0) {
+                ground[k][j].value = ground[k + 1][j].value;
+                ground[k + 1][j].value = 0;
+              }
+            }
+          }
+        }
+      }
+    } else if (direction.key === 'ArrowDown') {
+      for (j = 0; j < data.site.siteMetadata.size; j += 1) {
+        for (i = data.site.siteMetadata.size - 2; i > 0; i -= 1) {
+          if (ground[i][j].value !== 0) {
+            for (k = i + 1; k < data.site.siteMetadata.size; k += 1) {
+              if (ground[k - 1][j].value === ground[k][j].value && ground[k][j].isUsed === false) {
+                ground[k][j].value *= 2;
+                ground[k][j].isUsed = true;
+                ground[k - 1][j].value = 0;
+                break;
+              } else if (ground[k][j].value === 0) {
+                ground[k][j].value = ground[k - 1][j].value;
+                ground[k - 1][j].value = 0;
+              }
+            }
+          }
+        }
+      }
+    }
+    console.log(" ");
+    for (i = 0; i < data.site.siteMetadata.size; i += 1) {
+      console.log(i + ":  ", ground[i][0].value, ground[i][1].value, ground[i][2].value, ground[i][3].value);
+      for (j = 0; j < data.site.siteMetadata.size; j += 1) {
+        board[i][j] = 0;
+        board[i][j] = ground[i][j].value;
+      }
+    }
+    console.log(" ");
+  }
+
+  newTile = () => {
```

Comment:
czy `newTile` ustawia nowy klocek na planszy? Jeżeli tak to może nazwać funkcję `placeNewTile`?